sunpy/net/attr.py

```diff
@@ -500,7 +498,7 @@ def __hash__(self):
     def __eq__(self, other):
         if not isinstance(other, self.__class__):
             return False
-        return self.attrs == other.attrs
+        return set(self.attrs) == set(other.attrs)
```

Comment:
Does this not change the results?

Reply:
No, it doesn't.

Reply:
Then why does it need to be a set?

Reply:
I was trying to make more consistent the syntax over all the iterables:
```python
>>> set((1, 2)) == set([1, 2])
True
>>> (1, 2) == [1, 2]
False
```
I can't be sure of what is passed into attrs so set is an additional check.

Reply:
Then why not for the other eq checks?

Reply:
I don't remember exactly, but I feel that the fact that `attrs` has not type checking may raise some edge cases and backfire, e.g.:
```python
>>> "quale" == "equal"
False
>>> set("quale") == set("equal")
True
```
Anyway I'm reverting this.